include/boost/gil/algorithm.hpp

```diff
@@ -777,13 +814,19 @@ void uninitialized_copy_aux(It1 first1, It1 last1, It2 first2, std::false_type)
 template <typename View1, typename View2>
 void uninitialized_copy_pixels(View1 const& view1, View2 const& view2)
 {
-    using is_planar = std::integral_constant<bool, is_planar<View1>::value && is_planar<View2>::value>;
+    using copy_planarity_condition = detail::copy_planarity_condition;
+    using vxv_planarity = std::integral_constant<copy_planarity_condition,
+                                    !is_planar<View2>::value ?
+                                        copy_planarity_condition::mixed_2_interleaved :
+                                        (is_planar<View1>::value ?
+                                            copy_planarity_condition::planar_2_planar:
+                                            copy_planarity_condition::interleaved_2_planar)>;
```

Comment:
As I mentioned earlier, we try to lean the code to left, not to right (trying to keep line length limit to 100 characters) - there is still plenty of places where the code has not been re-formatted though.

The `<` and `>` can work like `{` and `}` 

```cpp
using vxv_planarity = std::integral_constant
    <
        copy_planarity_condition,
        !is_planar<View2>::value
            ? copy_planarity_condition::mixed_2_interleaved
            : (is_planar<View1>::value 
                ? copy_planarity_condition::planar_2_planar
                : copy_planarity_condition::interleaved_2_planar)
    >;
```

following formatting like here, for example:

https://github.com/boostorg/gil/blob/422ca82fe58d5aed90f86a2b043a402a2dc48a53/include/boost/gil/algorithm.hpp#L434-L454